### ISSUE TITLE
Initial load

### ISSUE THREAD

**Issue by dalsscarlon** [henrychavez/nativescript-bottom-navigation#8]
It's keep giving error when login into the app.
ERROR TypeError: undefined is not an object (evaluating 'this.nativeView.selectItemAnimated')

- iOS 11.2.5 iphone 6 
- tns 3.4.0
- tns ios 3.4.1
```
{
  "description": "xxx",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "org.nativescript.xxxx",
    "tns-android": {
      "version": "3.4.0"
    },
    "tns-ios": {
      "version": "3.4.1"
    }
  },
  "dependencies": {
    "@angular/animations": "~4.4.1",
    "@angular/common": "~4.4.1",
    "@angular/compiler": "~4.4.1",
    "@angular/core": "~4.4.1",
    "@angular/forms": "~4.4.1",
    "@angular/http": "~4.4.1",
    "@angular/platform-browser": "~4.4.1",
    "@angular/router": "~4.4.1",
    "email-validator": "^1.1.1",
    "nativescript-angular": "~4.4.0",
    "nativescript-bottom-navigation": "^1.0.2",
    "nativescript-couchbase": "^1.0.18",
    "nativescript-floatingactionbutton": "4.1.3",
    "nativescript-fontawesome": "1.0.0",
    "nativescript-geolocation": "^4.2.2",
    "nativescript-gradient": "^2.0.1",
    "nativescript-iqkeyboardmanager": "1.1.0",
    "nativescript-photoviewer": "1.2.0",
    "nativescript-pro-ui": "3.3.0",
    "nativescript-theme-core": "~1.0.2",
    "reflect-metadata": "~0.1.8",
    "rxjs": "~5.4.2",
    "tns-core-modules": "^3.4.0",
    "zone.js": "^0.8.19"
  },
  "devDependencies": {
    "babel-traverse": "6.26.0",
    "babel-types": "6.26.0",
    "babylon": "6.18.0",
    "lazy": "1.0.11",
    "nativescript-dev-typescript": "~0.5.0",
    "typescript": "~2.4.2"
  }
}
```

The issue is when i call selectTab()

**Comment by henrychavez** [comment]
Hi,
Can u pealse send me an example of how are you using the selectTab and when?

On Jan 31, 2018 9:21 PM, "Dalia Carlón" <notifications@github.com> wrote:

It's keep giving error when login into the app.
ERROR TypeError: undefined is not an object (evaluating 'this.nativeView.
selectItemAnimated')

   - iOS 11.2.5 iphone 6
   - tns 3.4.0
   - tns ios 3.4.1

{
  "description": "xxx",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "org.nativescript.xxxx",
    "tns-android": {
      "version": "3.4.0"
    },
    "tns-ios": {
      "version": "3.4.1"
    }
  },
  "dependencies": {
    "@angular/animations": "~4.4.1",
    "@angular/common": "~4.4.1",
    "@angular/compiler": "~4.4.1",
    "@angular/core": "~4.4.1",
    "@angular/forms": "~4.4.1",
    "@angular/http": "~4.4.1",
    "@angular/platform-browser": "~4.4.1",
    "@angular/router": "~4.4.1",
    "email-validator": "^1.1.1",
    "nativescript-angular": "~4.4.0",
    "nativescript-bottom-navigation": "^1.0.2",
    "nativescript-couchbase": "^1.0.18",
    "nativescript-floatingactionbutton": "4.1.3",
    "nativescript-fontawesome": "1.0.0",
    "nativescript-geolocation": "^4.2.2",
    "nativescript-gradient": "^2.0.1",
    "nativescript-iqkeyboardmanager": "1.1.0",
    "nativescript-photoviewer": "1.2.0",
    "nativescript-pro-ui": "3.3.0",
    "nativescript-theme-core": "~1.0.2",
    "reflect-metadata": "~0.1.8",
    "rxjs": "~5.4.2",
    "tns-core-modules": "^3.4.0",
    "zone.js": "^0.8.19"
  },
  "devDependencies": {
    "babel-traverse": "6.26.0",
    "babel-types": "6.26.0",
    "babylon": "6.18.0",
    "lazy": "1.0.11",
    "nativescript-dev-typescript": "~0.5.0",
    "typescript": "~2.4.2"
  }
}

The issue is when i call selectTab()

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/henrychavez/nativescript-bottom-navigation/issues/8>,
or mute the thread

**Comment by dalsscarlon** [comment]
I have a main.ts that is the hole "frame/ container" of the project, when login is redirecting to `this.router.navigate(["/main/prm"], { clearHistory: true });`, that execute the main.ts: 

```
import { Component, OnInit,OnDestroy,ElementRef,ViewChild } from "@angular/core";
import { BottomNavigation, OnTabSelectedEventData, BottomNavigationTab } from 'nativescript-bottom-navigation';
import { Router} from "@angular/router";
import { Page} from "ui/page";
import {ConnectionInfo} from "../../services/connectivity.service";
import { isAndroid } from "platform";
import { AppComponent } from "../../app.component";

@Component(
  {
    selector: "main",
    moduleId: module.id,
    templateUrl: "main.component.html"
  }
)
export class mainComponent implements OnInit {
  @ViewChild("bottomNav") bottomNavChild:ElementRef;
  bottomNav:BottomNavigation;
  pestanhaInicial:number=1;
  
  rutas:any[] =
            [{ruta:"/main/vtc",color:"#4EDFC2"},
            {ruta:"/main/prm",color:"#FD7E7E"},
            {ruta:"/main/cnd",color:"#2196F3"},
            {ruta:"/main/news",color:"#F9883B"},
            {ruta:"/main/tips",color:"#FFB901"}]
  online:boolean = true;
  actual : number=1;        
            
  constructor(private router:Router, private page: Page, private connectionInfo: ConnectionInfo, private component: AppComponent){}
  ngOnInit(){
    
    this.bottomNav = this.bottomNavChild.nativeElement;
    if(isAndroid){
      this.bottomNav.inactiveColor="lightgray";
      this.bottomNav.backgroundColor="white";
      this.bottomNav.activeColor=this.rutas[this.pestanhaInicial].color;
      this.bottomNav.selectedTabIndex = this.pestanhaInicial;
    }
    else{
      this.bottomNav.selectTab(this.pestanhaInicial);
    }
        
    this.router.navigate(['/main/prm'],{replaceUrl:true});
  }

  onBottomNavigationTabSelected(args: OnTabSelectedEventData){
    this.actual = args.newIndex;
    let destiny:string = this.rutas[this.actual].ruta;
    let color:string = this.rutas[this.actual].color;
    this.router.navigate([destiny],{replaceUrl:true});
    this.bottomNavChild.nativeElement.activeColor=color;

    this.component.checkToken();
  }
}
```
I think it is because the bottomnavigation isn't build yet.

**Comment by dalsscarlon** [comment]
If i don't explain well, i could explain it in spanish

**Comment by henrychavez** [comment]
@dalsscarlon sorry the delay I didn't receive a notification.

The problems is when you call the selectTab the BottomNavigationBar is not created yet.

But I see that you are using the selectedTabIndex for android and it's working, so I'll update the plugin to do the same on iOS.

I let you now when is ready so you can have 
```
this.bottomNav.inactiveColor="lightgray";
this.bottomNav.backgroundColor="white";
this.bottomNav.activeColor=this.rutas[this.pestanhaInicial].color;
this.bottomNav.selectedTabIndex = this.pestanhaInicial;
```

without the if statement

**Comment by henrychavez** [comment]
The bug was fixed in the version 1.1.3 of the plugin. :D
